src/python/pants/backend/python/subsystems/debugpy.py

```diff
@@ -42,13 +58,13 @@ def _get_main_spec_args(main: MainSpecification) -> tuple[str, ...]:
             return ("-m", main.module)
 
         assert isinstance(main, ConsoleScript)
-        # NB: This is only necessary while debugpy supports Py 3.7, as `importlib.metadata` was
-        # added in Py 3.8 and would allow us to resolve the console_script using just the stdlib.
         return (
             "-c",
             (
-                "from pex.third_party.pkg_resources import working_set;"
```

Comment:
How were you expecting you got the pex dependency previously? AFAICT it's nowhere listed as a dep for the DebugPy subsystem. Is it the case you were just getting lucky in Pants because Pants itself does have a 3rdparty dep on Pex?